src/rpc/blockchain.cpp

```diff
@@ -782,6 +782,24 @@ static RPCHelpMan getblock()
     };
 }
 
+//! Return height of highest block that has been pruned, or std::nullopt if no blocks have been pruned
```

Comment:
In commit "refactor, blockstorage: Generalize GetFirstStoredBlock" (dca1ca1f56d8e63929eb7aea64f8fb3e01752357)

I think the code would be simpler (and have fewer cases to think) about if this function just returned -1 when no blocks were pruned, instead of nullopt:

```diff
--- a/src/rpc/blockchain.cpp
+++ b/src/rpc/blockchain.cpp
@@ -782,8 +782,8 @@ static RPCHelpMan getblock()
     };
 }
 
-//! Return height of highest block that has been pruned, or std::nullopt if no blocks have been pruned
-static std::optional<int> GetPruneHeight(BlockManager& blockman, const CChain& chain) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
+//! Return height of highest block that has been pruned, or -1 if no blocks have been pruned
+int GetPruneHeight(BlockManager& blockman, const CChain& chain) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) {
     AssertLockHeld(::cs_main);
 
     const CBlockIndex* chain_tip{chain.Tip()};
@@ -794,7 +794,7 @@ static std::optional<int> GetPruneHeight(BlockManager& blockman, const CChain& c
     const auto& first_unpruned{*Assert(blockman.GetFirstBlock(*chain_tip, /*status_mask=*/BLOCK_HAVE_DATA))};
     if (!first_unpruned.pprev) {
        // No block before the first unpruned block means nothing is pruned.
-       return std::nullopt;
+       return -1;
     }
     // Block before the first unpruned block is the last pruned block.
     return first_unpruned.pprev->nHeight;
@@ -852,7 +852,7 @@ static RPCHelpMan pruneblockchain()
     }
 
     PruneBlockFilesManual(active_chainstate, height);
-    return GetPruneHeight(chainman.m_blockman, active_chain).value_or(-1);
+    return GetPruneHeight(chainman.m_blockman, active_chain);
 },
     };
 }
@@ -1305,8 +1305,7 @@ RPCHelpMan getblockchaininfo()
     obj.pushKV("size_on_disk", chainman.m_blockman.CalculateCurrentUsage());
     obj.pushKV("pruned", chainman.m_blockman.IsPruneMode());
     if (chainman.m_blockman.IsPruneMode()) {
-        const auto prune_height{GetPruneHeight(chainman.m_blockman, active_chainstate.m_chain)};
-        obj.pushKV("pruneheight", prune_height ? prune_height.value() + 1 : 0);
+        obj.pushKV("pruneheight", GetPruneHeight(chainman.m_blockman, active_chainstate.m_chain) + 1);
 
         const bool automatic_pruning{chainman.m_blockman.GetPruneTarget() != BlockManager::PRUNE_TARGET_MANUAL};
         obj.pushKV("automatic_pruning",  automatic_pruning);
```